src/main/java/com/synopsys/integration/alert/workflow/scheduled/PurgeTask.java

```diff
@@ -130,20 +128,19 @@ private void purgeNotifications() {
 
     private void purgeSystemMessages() {
         try {
-            Date date = createDate();
+            OffsetDateTime date = createDate();
             List<SystemMessageModel> messages = systemMessageUtility.getSystemMessagesBefore(date);
             systemMessageUtility.deleteSystemMessages(messages);
         } catch (Exception ex) {
             logger.error("Error purging system messages", ex);
         }
     }
 
-    public Date createDate() {
-        ZonedDateTime zonedDate = ZonedDateTime.now();
-        zonedDate = zonedDate.minusDays(dayOffset);
-        zonedDate = zonedDate.withZoneSameInstant(ZoneOffset.UTC);
-        zonedDate = zonedDate.withHour(0).withMinute(0).withSecond(0).withNano(0);
-        return Date.from(zonedDate.toInstant());
+    // TODO give this method a more descriptive name
+    public OffsetDateTime createDate() {
+        return OffsetDateTime.now()
```

Comment:
There are a few places we create a Date and substract an offset. Shouldn't we have a method in DateUtils to do this?

Reply:
I see this used in `NotificationManager.findByCreatedAtBeforeDayOffset` and `PurgeTask` but the `NotificationMananger` interface I don't find any usage of the method outside of tests.  It may have been used in processing in the past but may not be used anymore.  We might want to TODO or FIXME to remove the method in `NotificationManager`